LittleToySourceGenerator/LittleToySourceGenerator.cs

```diff
@@ -1,30 +1,90 @@
-﻿using Microsoft.CodeAnalysis;
-using Microsoft.CodeAnalysis.CSharp.Syntax;
-using Microsoft.CodeAnalysis.Text;
-using System;
+﻿using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Reflection;
 using System.Text;
+using CsCodeGenerator;
+using CsCodeGenerator.Enums;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
 
 namespace LittleToySourceGenerator;
 
 [Generator]
 public class Generator : ISourceGenerator
 {
-    public bool GenerateEventData { get; set; } = true;
+    public bool EnableEventDataGeneration { get; set; } = true;
     public bool GenerateEventSystem { get; set; } = true;
     public bool GenerateEventViewInterface { get; set; } = true;
     public bool GenerateReadWriteEcs { get; set; } = true;
 
+    public const string ComponentDirtyEventAttributeType = "ComponentDirtyEvent";
+
+    public const string CodeGenNetComponentAttributeType = "CodeGenNetComponent";
+
+    public const string MarkDirtyAttributeType = "MarkDirty";
+
+    public const string SyncFieldAttributeType = "SyncField";
+
+    public const string ComponentRemovedEventAttributeType = "ComponentRemovedEvent";
+
+    public const string ComponentAddedEventAttributeType = "ComponentAddedEvent";
+
+    private const string FileHeader = @"// <auto-generated>
+// Code generated by LittleToy Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is
+// regenerated.
+// </auto-generated>
+#nullable enable
+#pragma warning disable 1591";
+
+    private static Dictionary<string, string> _systemToDotsnetTypeDictionary = new Dictionary<string, string>()
```

Comment:
Would like that somebody review namespaces for these types.

Reply:
Namespaces below are representing types. Previously we use `typeof(byte)` and that was mapped to type identity `System.Byte` by compiler. Right now when I'm inside Roslyn I need to know name of the type. Also I cannot have dependency on `DOTSNET` and `Unity` assemblies, so I have to spell full names of types below.